### ISSUE TITLE
Application startup exception

### ISSUE THREAD

**Issue by mcselasvegas** [PiranhaCMS/piranha.core#1204]
I am trying to get the mvc example to run and get the following error. I assume it has something to do with the database, but wonder if this is a known issue....

      Application startup exception
System.ComponentModel.DataAnnotations.ValidationException: The InternalId field must be unique
   at Piranha.Services.SiteService.SaveAsync(Site model) in C:\inetpub\wwwroot\Vatrigo.Com\Piranha.Core-dev\piranha.core-master\core\Piranha\Services\SiteService.cs:line 279
   at MvcWeb.Seed.RunAsync(IApi api) in C:\inetpub\wwwroot\Vatrigo.Com\Piranha.Core-dev\piranha.core-master\examples\MvcWeb\Seed.cs:line 40
   at MvcWeb.Startup.Configure(IApplicationBuilder app, IWebHostEnvironment env, IApi api) in C:\inetpub\wwwroot\Vatrigo.Com\Piranha.Core-dev\piranha.core-master\examples\MvcWeb\Startup.cs:line 139
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.AspNetCore.Hosting.ConfigureBuilder.Invoke(Object instance, IApplicationBuilder builder)
   at Microsoft.AspNetCore.Hosting.ConfigureBuilder.<>c__DisplayClass4_0.<Build>b__0(IApplicationBuilder builder)
   at Microsoft.AspNetCore.Hosting.GenericWebHostBuilder.<>c__DisplayClass13_0.<UseStartup>b__2(IApplicationBuilder app)
   at Microsoft.AspNetCore.Mvc.Filters.MiddlewareFilterBuilderStartupFilter.<>c__DisplayClass0_0.<Configure>g__MiddlewareFilterBuilder|0(IApplicationBuilder builder)
   at Microsoft.AspNetCore.Server.IISIntegration.IISSetupFilter.<>c__DisplayClass4_0.<Configure>b__0(IApplicationBuilder app)
   at Microsoft.AspNetCore.HostFilteringStartupFilter.<>c__DisplayClass0_0.<Configure>b__0(IApplicationBuilder app)
   at Microsoft.AspNetCore.Hosting.GenericWebHostService.StartAsync(CancellationToken cancellationToken)

**Comment by tidyui** [comment]
The MVC example from the repository or the MVC template?

**Comment by mcselasvegas** [comment]
I should have clarified that... I use the MVC example from the repository downloaded on 05/58/2020. I also made the changes explained here to use MSSQL https://piranhacms.org/docs/architecture/databases/sql-server. The tables are created as expected upon initial start of the application. But that is when the error occurs... BTW it also happens with the out-of-box SQLlite setup.

On more thing... I had to change the web.config hostingModel from inprocess to outofprocess for the application to start in the first place.

**Comment by tidyui** [comment]
Hi there. I can't reproduce with the code from the master branch, the MVC example starts and runs just as expected. The error message you give usually happens when the database has been partially created and it tries to create the default data again (that's why a duplicate key is inserted in the Site table).

Maybe you could try to delete the database and start over?

**Comment by tidyui** [comment]
Did you work this out @mcselasvegas?

**Comment by mcselasvegas** [comment]
I did... just for future readers.

It turns out that when I published the MvcWeb Example from the repository it was missing the seed folder and therefor did not complete the install. This is when I turned on logging in the web.config and received the `The InternalId field must be unique` error. Once I deleted the database copied the seed folder over to my published application and ran the application again, it worked!

Thank you for your assistance - and this great application!
Stay Safe - Oliver
